rust/utils/run-locally/src/main.rs

```diff
@@ -425,24 +431,11 @@ fn main() -> ExitCode {
     let loop_start = Instant::now();
     // give things a chance to fully start.
     sleep(Duration::from_secs(5));
-    let mut kathy_done = false;
     while RUNNING.fetch_and(true, Ordering::Relaxed) {
-        if !kathy_done {
-            // check if kathy has finished
-            match state.kathy.as_mut().unwrap().try_wait().unwrap() {
-                Some(s) if s.success() => {
-                    kathy_done = true;
-                }
-                Some(_) => {
-                    return ExitCode::from(1);
-                }
-                None => {}
-            }
-        }
         if ci_mode {
             // for CI we have to look for the end condition.
             let num_messages_expected = (kathy_messages / 2) as u32 * 2;
-            if kathy_done && termination_invariants_met(num_messages_expected) {
+            if termination_invariants_met(num_messages_expected) {
```

Comment:
nit: Update log messages below to reflect the removal of the kathy finishing requirement